tests/reftests/pin.test

```diff
@@ -1,7 +1,811 @@
+User metadata backed up in ${BASEDIR}/OPAM/backup/nip-of.bak
+The following actions will be performed:
+=== upgrade 1 package
+  - upgrade nip-of dev1 to dev2 (pinned)
+
+<><> Processing actions <><><><><><><><><><><><><><><><><><><><><><><><><><><><>
+-> removed   nip-of.dev1
+-> installed nip-of.dev2
+Done.
+### opam show nip-of --field=version
+dev2
+### sh add-pin.sh nip-of.dev3
+### git -C ./nip-of add nip-of.dev3.t
+### git -C ./nip-of commit -qm "add install file"
+### opam pin ./nip-of
+[NOTE] Package nip-of is already pinned to git+file://${BASEDIR}/nip-of#master (version dev2).
+nip-of is now pinned to git+file://${BASEDIR}/nip-of#master (version dev3)
+
+The following actions will be performed:
+=== upgrade 1 package
+  - upgrade nip-of dev2 to dev3 (pinned)
+
+<><> Processing actions <><><><><><><><><><><><><><><><><><><><><><><><><><><><>
+-> retrieved nip-of.dev3  (no changes)
+-> removed   nip-of.dev2
+-> installed nip-of.dev3
+Done.
+### opam show nip-of --field=version
+dev3
+### : only installs
+### sh add-pin.sh nip-fo.dev1 git
+### git -C ./nip-fo rm nip-fo.opam --cached -q
+### git -C ./nip-fo commit -qm "remove opam file"
+### opam install ./nip-fo
+Package nip-fo does not exist, create as a NEW package? [y/n] y
+nip-fo is now pinned to git+file://${BASEDIR}/nip-fo#master (version dev1)
+The following actions will be performed:
+=== install 1 package
+  - install nip-fo dev1 (pinned)
+
+<><> Processing actions <><><><><><><><><><><><><><><><><><><><><><><><><><><><>
+-> retrieved nip-fo.dev1  (git+file://${BASEDIR}/nip-fo#master)
+-> installed nip-fo.dev1
+Done.
+### opam show nip-fo --field=version
+dev1
+### sh add-pin.sh nip-fo.dev2 git
+### opam install ./nip-fo
+[nip-fo.dev1] synchronised (git+file://${BASEDIR}/nip-fo#master)
+[nip-fo] Conflicting update of the metadata from git+file://${BASEDIR}/nip-fo#master.
+Override files in ${BASEDIR}/OPAM/opamfiles/.opam-switch/overlay/nip-fo (there will be a backup)? [y/n] y
+User metadata backed up in ${BASEDIR}/OPAM/backup/nip-fo.bak
+The following actions will be performed:
+=== upgrade 1 package
+  - upgrade nip-fo dev1 to dev2 (pinned)
+
+<><> Processing actions <><><><><><><><><><><><><><><><><><><><><><><><><><><><>
+-> removed   nip-fo.dev1
+-> installed nip-fo.dev2
+Done.
+### opam show nip-fo --field=version
+dev2
+### sh add-pin.sh nip-fo.dev3
+### git -C ./nip-fo add nip-fo.dev3.t
+### git -C ./nip-fo commit -qm "add install file"
+### opam install ./nip-fo
+[NOTE] Ignoring uncommitted changes in ${BASEDIR}/nip-fo (`--working-dir' not specified or specified with no argument).
+[nip-fo.dev2] synchronised (git+file://${BASEDIR}/nip-fo#master)
+The following actions will be performed:
+=== recompile 1 package
+  - recompile nip-fo dev2 (pinned)
+
+<><> Processing actions <><><><><><><><><><><><><><><><><><><><><><><><><><><><>
+-> removed   nip-fo.dev2
+-> installed nip-fo.dev2
+Done.
+### opam show nip-fo --field=version
+dev2
```

Comment:
Is it a bug?

Reply:
yes it is a known bug, i've seen it several times but i can't find the issue that i used to redirect people to when they got that issue

Reply:
https://github.com/ocaml/opam/issues/5567#issuecomment-2139447699 maybe?